client/lib/domains/registration/availability-messages.js

```diff
@@ -308,6 +308,27 @@ function getAvailabilityNotice(
 
 		case domainAvailability.MAPPABLE:
 			if ( isForTransferOnly ) {
+				if ( errorData?.cannot_transfer_due_to_unsupported_premium_tld ) {
+					message = translate(
+						"This domain cannot be transferred to WordPress.com because it's premium and currently we don't support premium transfers for the %(tld)s TLD. It can be connected instead. {{a}}Learn More.{{/a}}",
```

Comment:
Not sure "More" should be capitalised here, but may I recommend tidying up this string to be more consistent with the other errors anyway? 

```suggestion
						"Premium domains ending with %(tld)s cannot be transferred to WordPress.com. Please connect your domain instead. {{a}}Learn more.{{/a}}",
```

Reply:
@Aurorum good catch, and I like the suggestion. I'll update the copy 👍